app/src/main/java/kth/jjve/xfran/EventPlanActivity.java

```diff
@@ -81,29 +91,79 @@ protected void onCreate(Bundle savedInstanceState) {
+            int mMonth = calendar.get(Calendar.MONTH); // ADD +1 to get actual month!!!
+            int mDay = calendar.get(Calendar.DAY_OF_MONTH);
+
+            datePickerDialog = new DatePickerDialog(EventPlanActivity.this,
+                    (view, year, month, day) -> {
+                        // set day of month , month and year value in the edit text
+                        String date = day + "/"  + (month + 1) + "/" + year;
+                        eventDateInput.setText(date);
+                    }, mYear, mMonth, mDay);
+            datePickerDialog.getDatePicker().setFirstDayOfWeek(Calendar.MONDAY);
+            datePickerDialog.show();
+        });
+
+        /*------ TIME PICKER DIALOG -----*/
+        // TODO: see if can be implemented in method?
+        eventStartTimeInput.setOnClickListener(v -> {
+            Calendar mCurrentTime = Calendar.getInstance();
+            int hour = mCurrentTime.get(Calendar.HOUR_OF_DAY);
+            int minute = mCurrentTime.get(Calendar.MINUTE);
+            TimePickerDialog timePickerDialog;
+            timePickerDialog = new TimePickerDialog(EventPlanActivity.this, (view, hourOfDay, minuteOfHour) -> {
+                @SuppressLint("DefaultLocale") String s_time = String.format("%02d:%02d", hourOfDay, minuteOfHour);
+                eventStartTimeInput.setText(s_time);
+            }, hour, minute, true);//Yes 24 hour time
+            timePickerDialog.show();
+        });
+
+        eventEndTimeInput.setOnClickListener(v -> {
+            Calendar mCurrentTime = Calendar.getInstance();
+            int hour = mCurrentTime.get(Calendar.HOUR_OF_DAY);
+            int minute = mCurrentTime.get(Calendar.MINUTE);
+            TimePickerDialog timePickerDialog;
+            timePickerDialog = new TimePickerDialog(EventPlanActivity.this, (view, hourOfDay, minuteOfHour) -> {
+                @SuppressLint("DefaultLocale") String s_time = String.format("%02d:%02d", hourOfDay, minuteOfHour);
+                eventEndTimeInput.setText(s_time);
+            }, hour, minute, true);//Yes 24 hour time
+            timePickerDialog.show();
+        });
+
     }
 
     private void setEventInApp(){
         // method to obtain the events name and start/end time
         if (mWorkout == null){
             s_eventName = eventNameET.getText().toString();
         }
-        startTime = eventStartTimeEdit.getText().toString();
-        stopTime = eventEndTimeEdit.getText().toString();
+        s_eventDate = eventDateInput.getText().toString();
+        startTime = eventStartTimeInput.getText().toString();
+        stopTime = eventEndTimeInput.getText().toString();
 
         if (TextUtils.isEmpty(s_eventName)){
             eventNameET.setError("Event name is required");
             return;
         }
 
-        //TODO make safety feature if time format wrong pretty
+        if (TextUtils.isEmpty(s_eventDate)){
+            eventDateInput.setError("Date is required");
+            return;
```

Comment:
Can't we combine all the errors into one error message saying "required"?
I think it would be clear on the user interface, plus it would clear up the backend